connectors/sources/github.py

```diff
@@ -2142,24 +2168,14 @@ async def get_docs(self, filtering=None):
                         else:
                             yield file_document, None
 
-                if path := rule["filter"].get(ObjectType.PATH.value):
-                    async for (
-                        file_document,
-                        attachment_metadata,
-                    ) in self._fetch_files_by_path(repo_name=repo_name, path=path):
-                        if file_document["type"] == FILE:
-                            attachment_metadata["url"] = attachment_metadata["git_url"]
-                            yield (
-                                file_document,
-                                partial(
-                                    self.get_content, attachment=attachment_metadata
-                                ),
-                            )
-                        else:
-                            yield file_document, None
+                self._mark_repo_as_processed(repo_name)
         else:
             async for repo in self._fetch_repos():
-                if self.is_previous_repo(repo["nameWithOwner"]):
+                repo_name = repo.get("nameWithOwner")
+                if self.is_previous_repo(repo_name):
+                    continue
```

Comment:
I wonder if this is obsolete if we are already tracking the processed repos?

Reply:
Hmmm you are right, good catch.

I really don't know where this is coming from, so anxious to delete it cause I'm not sure how to test it. I'll give it a look and see if I can test, if not I'll just add a #TODO and leave it be